Character.cpp

```diff
@@ -86,8 +125,60 @@ int Character::takeDamage(int damage)
     return hitPoints;
 }
 
+void Character::levelUpHitPoints()
+{
+    if(*initialHitPoints.get())
```

Comment:
`get()` returns a pointer, which may be nullptr.  Then you're dereferencing it, which may be a nullptr dereference.
Then, because `initialHitPoints` owns an `int`, you're basically checking if the owned value is not zero.
All of this indicates you do not understand how to check if a raw pointer or `unique_ptr<>` instance is safe to use or not.

Reply:
ok